File: src/main/resources/output/IFC4X3_csharp/Types/IfcCoordinateReferenceSystemSelect.cs
// SELECT TYPE IfcCoordinateReferenceSystemSelect
public class IfcCoordinateReferenceSystemSelect {
    public object Value { get; }

    public IfcCoordinateReferenceSystemSelect(object value) {
        if (!(value is IfcCoordinateReferenceSystem || value is IfcGeometricRepresentationContext))
        {
            throw new ArgumentException("Value must be one of: IfcCoordinateReferenceSystem, IfcGeometricRepresentationContext");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}

File: src/main/resources/output/IFC4X3_vbnet/Types/IfcCoordinateReferenceSystemSelect.vb
' SELECT TYPE IfcCoordinateReferenceSystemSelect
' Options:
' - IfcCoordinateReferenceSystem
' - IfcGeometricRepresentationContext
Public Class IfcCoordinateReferenceSystemSelect
    Public ReadOnly Property Value As Object

    Public Sub New(value As Object)
        If Not (TypeOf value Is IfcCoordinateReferenceSystem OrElse TypeOf value Is IfcGeometricRepresentationContext) Then
            Throw New ArgumentException("Value must be one of: IfcCoordinateReferenceSystem, IfcGeometricRepresentationContext")
        End If
        Me.Value = value
    End Sub

    Public Function GetValueAs(Of T)() As T
        Return CType(Value, T)
    End Function
End Class
